### PR TITLE
Improve given_function_called

### LINKED ISSUE #1682: Method `given_function_called` should only take the function name into account.
# Bug report

## What's wrong

The method `given_function_called(node: ast.Call, to_check: Container[str]) -> str` in `logic.tree.functions` is described as a method that returns the name of the function being called in `node`, in case it is included in `to_check`. For example:
 ```python
# Let's imagine we are visiting the Call node in `print(123, 456)` stored in `node`
called_function = given_function_called(node, ['print'])
print(called_function)
# Prints `print`
# But, if we are visiting `datetime.timedelta(days=1)`
called_function = given_function_called(node, ['timedelta'])
print(called_function)
# Prints an empty string, as if `timedelta` had not been called. The way for it to be shown is:
called_function = given_function_called(node, ['datetime.timedelta'])
print(called_function)
# Prints `datetime.timedelta`
```

This is related to https://github.com/wemake-services/wemake-python-styleguide/pull/1676#discussion_r508471791

## How is that should be

```python
# If we are visiting `datetime.timedelta(days=1)`
called_function = given_function_called(node, ['timedelta'])
print(called_function)
# Prints `timedelta`
```

### PR DESCRIPTION
**Pull request by Jrryy**
# I have made things!

Added a parameter `split_modules: bool` to `given_function_called` in `wemake_python_styleguide.logic.tree.functions`. Defaults to False.

When False, behaves as always.
When True, it only checks that a function has been called, regardless of the modules it comes from. For example:

```python
import ast

module = ast.parse('datetime.timedelta(days=1)')
called_function = given_function_called(module.body[0].value, ['datetime.timedelta'], split_modules=False)
print(called_function) 
# Prints `datetime.timedelta`

called_function = given_function_called(module.body[0].value, ['timedelta'], split_modules=False)
print(called_function)
# Prints `` (empty string)

called_function = given_function_called(module.body[0].value, ['timedelta'], split_modules=True)
print(called_function)
# Prints `timedelta`
```

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #1682

### CODE REVIEW

File: wemake_python_styleguide/logic/tree/functions.py [hunk 1]
```diff
@@ -10,10 +10,17 @@
 )
 
 
-def given_function_called(node: Call, to_check: Container[str]) -> str:
+def given_function_called(
+    node: Call,
+    to_check: Container[str],
+    split_modules: bool = False,
```
Comment (sobolevn) [inline wemake_python_styleguide/logic/tree/functions.py:16]: Lets makeit `kwonly`: 
```python
*,
split_modules: bool = False,
```

File: wemake_python_styleguide/logic/tree/functions.py [hunk 2]
```diff
@@ -22,8 +29,25 @@ def given_function_called(node: Call, to_check: Container[str]) -> str:
     >>> given_function_called(module.body[0].value, ['adjust'])
     ''
 
+    >>> module = ast.parse('datetime.timedelta(days=1)')
```
Comment (sobolevn) [inline wemake_python_styleguide/logic/tree/functions.py:32]: We can remove all doctest, because they are replaced with new test cases.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks!

**Review by sobolevn — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ Semantic versioning in our case means:
 - WPS531: Forbids testing conditions to just return booleans when it is possible to simply return the condition itself
 - Forbids to use unsafe infinite loops
 - Forbids to use raw strings `r''` when not necessary
+- Add parameter to `given_function_called` so that it checks if a function has been called by its name regardless of the modules or classes it is in
```
Comment (sobolevn) [inline CHANGELOG.md:30]: Btw, we only put user-facing changes here. Som you can delete this line in the future PR.

**Comment by sobolevn** [comment]
Thanks again! 👍

File: CHANGELOG.md (hunk 3 above)
Comment (Jrryy) [reply CHANGELOG.md:30]: ah, gotcha

**Pull request merged**